tools/nni_cmd/nnictl_utils.py

```diff
@@ -373,6 +379,146 @@ def webui_url(args):
+                hdfs_client.delete(output_directory, recursive=True)
+
+def experiment_clean(args):
+    '''clean up the experiment data'''
+    nni_config = Config(get_config_filename(args))
+    while True:
+        print('INFO: clean up all data for experiment {0}'.format(args.id))
+        inputs = input('INFO: do you want to continue?[y/n]:')
+        if not inputs.lower() or inputs.lower() in ['n', 'no']:
+            print_normal('Exit!')
+            exit(0)
+        elif inputs.lower() not in ['y', 'n', 'yes', 'no']:
+            print_warning('please input Y or N!')
+        else:
+            break
+    #clean local data
+    home = str(Path.home())
+    local_dir = os.path.join(home, 'nni', 'experiments', str(args.id))
+    local_clean(local_dir)
+    platform = nni_config.get_config('experimentConfig').get('trainingServicePlatform')
+    if platform == 'remote':
+        remote_clean(nni_config)
+    elif platform == 'pai':
+        hdfs_clean(nni_config)
+    elif platform != 'local':
+        #TODO: support all platforms
+        print_warning('platform {0} clean up not supported yet!'.format(platform))
+        exit(0)
+    experiment_config = Experiments()
+    experiment_config.remove_experiment(args.id)
+    print_normal('Success!') 
+
+def get_platform_dir(experiment_dict, platform):
+    '''get directories in platform'''
+    dir_list = []
+    for key in experiment_dict.keys():
+        if platform != experiment_dict[key]['platform']:
+            continue
+        file_name = experiment_dict[key].get('fileName')
+        nni_config = Config(file_name)
+        if platform == 'remote':
+            machine_list = nni_config.get_config('experimentConfig').get('machineList')
+            for machine in machine_list:
+                host = machine.get('ip')
+                port = machine.get('port')
+                remote_dir = '/' + '/'.join(['tmp', 'nni', 'experiments', nni_config.get_config('experimentId')])
+                dir_list.append(host + ':' + str(port) + remote_dir)
+        elif platform == 'pai':	
+            user_name = nni_config.get_config('experimentConfig').get('paiConfig').get('userName')	
+            full_path = '/' + '/'.join([user_name, 'nni', 'experiments', nni_config.get_config('experimentId')])
+            dir_list.append(full_path)
+            output_dir = nni_config.get_config('experimentConfig').get('paiConfig').get('outputDir')
+            dir_list.append(output_dir)
+        else:
+            print_normal('not supported platform!')
+            exit(1)
+    return dir_list
+
+def platform_clean(args):
+    '''clean up the experiment data'''
+    config_path = os.path.abspath(args.config)
+    if not os.path.exists(config_path):
+        print_error('Please set correct config path!')
+        exit(1)
+    experiment_config = get_yml_content(config_path)
+    platform = experiment_config.get('trainingServicePlatform')
```

Comment:
'trainingServicePlatform'? What is the spec for the config file?

Reply:
The config file is same as the yaml file to create an experiment, like
```
authorName: default
experimentName: example_mnist
trialConcurrency: 1
maxExecDuration: 1h
maxTrialNum: 10
#choice: local, remote, pai
trainingServicePlatform: local
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner
  #SMAC (SMAC should be installed through nnictl)
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  command: python3 mnist.py
  codeDir: .
  gpuNum: 0

```